app/fetch-did-web-doc.py

```diff
@@ -0,0 +1,142 @@
+import argparse
+import subprocess
+import tempfile
+from urllib.request import urlopen, Request
+from urllib.error import URLError, HTTPError
+import ssl
+import base64
+import json
+import time
+import logging
+
+CONNECT_TIMEOUT = 5
+HTTP_RETRIES = 5
+HTTP_DEFAULT_RETRY_AFTER = 1
+
+AFETCH_DIR = "/tmp/scitt"
+
+def fetch_unattested(url, nonce):
+    response = request(url)
+
+    result = json.dumps(
+        {
+            "url": url,
+            "nonce": nonce,
+            "status_code": response["status_code"],
+            "body": base64.b64encode(response["body"]).decode("utf-8"),
+        }
+    ).encode("utf-8")
+
+    return result
+
+
+def fetch_attested(url, nonce):
+    success = False
+    retries = HTTP_RETRIES
+    with tempfile.NamedTemporaryFile() as out_path:
+        args = [
+            f"{AFETCH_DIR}/afetch",
+            f"{AFETCH_DIR}/libafetch.enclave.so.signed",
+            out_path.name,
+            url,
+            nonce,
+        ]
+        logging.info(f"Starting {' '.join(args)}")
+        while not success and retries >= 0:
+            retries -= 1
+            try:
+                subprocess.run(
+                    args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True
+                )
+                success = True
+                break
+            except subprocess.CalledProcessError as e:
+                logging.error(f"afetch failed with code {e.returncode}: {e.stdout}")
+            except Exception as e:
+                logging.error(f"Unknown error: {e}")
+            if retries >= 0:
+                logging.info(f"Retrying in {HTTP_DEFAULT_RETRY_AFTER} seconds")
+                time.sleep(HTTP_DEFAULT_RETRY_AFTER)
+            else:
+                raise e
+        if success:
```

Comment:
Can this line ever be reached with `success = False`? It seems to me not, since when we always throw an exception when we run out of retries.

Then you don't even need the `success` variable

Reply:
Good point, once attested-fetch supports returning attested errors then probably we want to have similar HTTP status based retry logic as for the unattested case.